CommandLine/protos/Settings.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto2";
+package buffers.resources;
+
+import "options.proto";
+
+message Settings {
+  /// <<<-------- API -------->>>
+  optional string target_audio = 1;
+  optional string target_platform = 2;
+  optional string target_compiler = 3;
+  optional string target_graphics = 4;
+  optional string target_widgets = 5;
+  optional string target_collision = 6;
+  optional string target_network = 7;
+  repeated string extensions = 8;
+
+  /// <<<-------- COMPILER -------->>>
+  optional uint32 inherit_strings = 101;
```

Comment:
Right, like I said, I just did the gaps to anticipate maybe later wanting to rearrange it in the UI and the proto without making that a backwards-incompatible change. But that's also why I chose a flat structure, so we can drop the gaps in field numbers if you want?

Reply:
Nested messages can be used to group these fields in the actual UI, too. Field numbers don't otherwise matter, as the declaration order is also available via reflection.

Reply:
Alright: fa988b419064147102cf562b96d22e082c91e1ce

I decided to keep the groupings I had last night which is what you want. I suppose it's ok to rearrange in the UI, the UI doesn't have to be exactly the same as the proto. Also, we can change it as much as we want prior to anybody actually using the first release.